crates/bevy_winit/src/lib.rs

```diff
@@ -281,9 +281,7 @@ pub fn winit_runner(mut app: App) -> AppExit {
 
     let mut runner_state = WinitAppRunnerState::default();
 
-    // TODO: AppExit is effectively a u8 we could use a AtomicU8 here instead of a mutex.
-    let mut exit_status = Arc::new(Mutex::new(AppExit::Success));
-    let handle_exit_status = exit_status.clone();
+    let (exit_sender, exit_receiver) = sync_channel(1);
```

Comment:
```suggestion
    // Create a synchronous channel with a bounds of 1, since ideally only one exit code will be sent before exiting the app.
    let (exit_sender, exit_receiver) = sync_channel(1);
```

It's a bit wordy, so feel free to rephrase it!